synapseadmin/client.go

```diff
@@ -20,3 +22,42 @@ type Client struct {
 func (cli *Client) BuildAdminURL(path ...any) string {
 	return cli.BuildURL(mautrix.SynapseAdminURLPath(path))
 }
+
+func (cli *Client) SyncRoomWithContextAdmin(ctx context.Context, nextBatch string, roomID string, prevBatch string) (*mautrix.RespSyncRoomMessagesAdmin, error) {
+	resSyncAdmin, err := cli.FullAdminSyncRequest(ctx, ReqSyncAdmin{
+		from: nextBatch,
```

Comment:
/messages is not the same as sync. Only /sync is sync. There's no need for two methods and the method signature should be similar to the client API /messages wrapper (https://pkg.go.dev/maunium.net/go/mautrix#Client.Messages)